app/src/main/java/com/sunasterisk/a14day_challenge/ui/signup/SignUpActivity.kt

```diff
@@ -1,13 +1,72 @@
 package com.sunasterisk.a14day_challenge.ui.signup
 
-import androidx.appcompat.app.AppCompatActivity
+import android.content.Intent
 import android.os.Bundle
+import android.view.View
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
 import com.sunasterisk.a14day_challenge.R
+import com.sunasterisk.a14day_challenge.data.UserRepository
+import com.sunasterisk.a14day_challenge.data.local.DataBaseHandler
+import com.sunasterisk.a14day_challenge.data.local.UserLocalDataSource
+import com.sunasterisk.a14day_challenge.data.local.dao.UserDAOImp
+import com.sunasterisk.a14day_challenge.ui.home.HomeActivity
+import com.sunasterisk.a14day_challenge.ui.login.LoginActivity
+import kotlinx.android.synthetic.main.activity_sign_up.*
 
-class SignUpActivity : AppCompatActivity() {
+class SignUpActivity : AppCompatActivity(), View.OnClickListener,
+    SignUpContract.View {
+
+    lateinit var userRepository: UserRepository
+    lateinit var presenter: SignUpPresenter
```

Comment:
lateinit sinh ra để giải quyết cho bài toán dependence inject, những trường hợp còn lại nếu không cần thiết, sử dụng Nullable type 
trường hợp này có thể dùng lazy init (keyword: lazy) 
```suggestion
    private val userRepository by lazy {
        UserRepository(..) 
    }
```